### PR TITLE
Simplify CacheTagHelper.ProcessAsync

### PR DESCRIPTION
**Pull request by pranavkm**
Modify ProcessAsync to cause all tasks to fail when the ongoing GetChildContentAsync call that they're
awaiting on fails.

### CODE REVIEW

**Comment by pranavkm** [comment]
I'd recommend using https://github.com/aspnet/Mvc/pull/6506?w=1 to look at the actual diff.

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs [hunk 1]
```diff
@@ -63,81 +63,73 @@ public CacheTagHelper(IMemoryCache memoryCache, HtmlEncoder htmlEncoder) : base(
...
                 {
-                    Task<IHtmlContent> result;
+                    var tokenSource = new CancellationTokenSource();
 
-                    if (!MemoryCache.TryGetValue(cacheKey, out result))
-                    {
-                        var tokenSource = new CancellationTokenSource();
-
-                        // Create an entry link scope and flow it so that any tokens related to the cache entries
-                        // created within this scope get copied to this scope.
```
Comment (javiercn) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs]: leave the comment

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs [hunk 2]
```diff
@@ -63,81 +63,73 @@ public CacheTagHelper(IMemoryCache memoryCache, HtmlEncoder htmlEncoder) : base(
...
-            IHtmlContent content = null;
+            IHtmlContent content;
 
             if (Enabled)
             {
                 var cacheKey = new CacheTagKey(this, context);
 
-                while (content == null)
+                Task<IHtmlContent> result;
+                if (!MemoryCache.TryGetValue(cacheKey, out result))
```
Comment (javiercn) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs]: Can you reverse the if/else order? It's more intuitive the other way around

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs [hunk 3]
```diff
@@ -63,81 +63,73 @@ public CacheTagHelper(IMemoryCache memoryCache, HtmlEncoder htmlEncoder) : base(
...
 
-            IHtmlContent content = null;
+            IHtmlContent content;
 
             if (Enabled)
             {
                 var cacheKey = new CacheTagKey(this, context);
 
-                while (content == null)
+                Task<IHtmlContent> result;
```
Comment (javiercn) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs]: Can you inline result?

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs]: Yeah. I think it needs to be moved to the `CreateEntry` call.

**Comment by pranavkm** [comment]
🆙 📅

**Pull request merged**